src/shell/shell_cmds.cpp

```diff
@@ -2167,3 +2168,46 @@ void DOS_Shell::CMD_VER(char *args)
 		         dos.version.major, dos.version.minor);
 	}
 }
+
+void DOS_Shell::CMD_VOL(char *args)
+{
+	HELP("VOL");
+
+	uint8_t drive_index;
+	size_t len = 0;
+	if (args) {
+		StripSpaces(args);
+		len = strlen(args);
+	}
+	if (len > 0) {
+		if (len < 2 || args[1] != ':') {
+			SyntaxError();
+			return;
+		}
+		drive_index = toupper(*args) - 'A';
+	} else {
+		drive_index = DOS_GetDefaultDrive();
+	}
+	char drive_letter = drive_index + 'A';
+
+	reg_ah = 0x44;            // DOS IOCTL
+	reg_al = 0x0D;            // Generic block device request
+	reg_bl = drive_index + 1; // Drive (0 = default drive, A = 1, B = 2, etc.)  DOSBox drives start A = 0, so add 1.
+	reg_ch = 0x08;            // Device type (0x08 for block device)
+	reg_cl = 0x66;            // Get volume information (returns serial number, volume label, and filesystem type)
+	reg_dx = 0;               // Offset pointer for return information (DS:DX). Set to 0 for beginning of data segment (hopefully this doesn't step on anyone else's memory?)
+	dos.errorcode = 0;
+	CALLBACK_RunRealInt(0x21);
+	if (dos.errorcode) {
+		WriteOut(MSG_Get("SHELL_EXECUTE_DRIVE_NOT_FOUND"), drive_letter);
+		return;
+	}
+	char label[12];
+	uint32_t serial = mem_readd(Segs.phys[ds] + 2);
+	MEM_BlockRead(Segs.phys[ds] + 6, label, 11); // Read in non-null terminated string with spaces at the end.
```

Comment:
instead of the 11, `ARRAY_LEN(label) - 1`

Reply:
Can `Segs.phys[ds] + 6` be de-magic'd? 

(I'm assuming that's where the label resides?)

Reply:
Comment above line please.

Reply:
Applied the `ARRAY_LEN(label) - 1`, though I'm not sure what you mean by "de-magic" the memory address.  It's just a hard-coded offset where DOS stores the volume label.

https://stanislavs.org/helppc/int_21-69.html

https://github.com/dosbox-staging/dosbox-staging/blob/9fd0271d089457991b76525fa3e8352c1addf7be/src/dos/dos_ioctl.cpp#L201-L204